recipes/computer_vision/object_detection/model_server/object_detection_server.py

```diff
@@ -1,19 +1,30 @@
-from transformers import AutoImageProcessor, AutoModelForObjectDetection
+from transformers import AutoImageProcessor, AutoModelForObjectDetection, AutoConfig
+from huggingface_hub import snapshot_download
 from PIL import Image, ImageDraw
 from fastapi import FastAPI
 from pydantic import BaseModel
 import torch
 import base64
 import os
 import io
+import shutil
 
 
 app = FastAPI()
 model = os.getenv("MODEL_PATH", default="facebook/detr-resnet-101")
 revision = os.getenv("MODEL_REVISION", default="no_timm")
 
-processor = AutoImageProcessor.from_pretrained(model, revision=revision)
-model = AutoModelForObjectDetection.from_pretrained(model, revision=revision)
+if os.path.isfile(model):
+    snapshot_download(repo_id="facebook/detr-resnet-101",
```

Comment:
this should be 
```suggestion
    snapshot_download(repo_id=model,
```

Reply:
I don't agree as model is a path of a file

Reply:
Also this is both the model-name / file patch as thats how it gets downloaded from the snapshot download

Reply:
through a env variable or just as a Python constant  ?

Reply:
Through env variable `MODEL_NAME`

Reply:
https://github.com/containers/ai-lab-recipes/blob/0c88acbfb83b23d0440864e8890395f9e5acdfa6/recipes/computer_vision/object_detection/model_server/object_detection_server.py#L14

The string `"facebook/detr-resnet-101"` is the model path and the repo name on huggingface. The model file name is `pytorch_model.bin`. So when we download a snapshot of the repo with `snapshot_download()` we should generate a directory `facebook/detr-resnet-101` with a number of files including `pytorch_model.bin`, `config.json`, etc.

Say a user wanted to use [detr-resnet-50](https://huggingface.co/facebook/detr-resnet-50/tree/main) instead,  
they would change the `MODEL_PATH` env variable when starting the server to access the correct config and additional files for that model.